src/Tooltip2.tsx

```diff
@@ -0,0 +1,292 @@
+import React, {Children, useEffect, useRef, useState} from 'react'
+import Box from './Box'
+import {BetterSystemStyleObject, merge, SxProp} from './sx'
+import {useId} from './hooks/useId'
+import {isFocusable} from '@primer/behaviors/utils'
+
+export type Tooltip2Props = {
+  direction?: 'n' | 'ne' | 'e' | 'se' | 's' | 'sw' | 'w' | 'nw'
+  text?: string
+  noDelay?: boolean
+  align?: 'left' | 'right'
+  wrap?: boolean
+  type?: 'label' | 'description'
+} & SxProp
+
+export type TriggerPropsType = {
+  'aria-describedby'?: string
+  'aria-labelledby'?: string
+  'aria-label'?: string
+  onFocus?: (event: React.FocusEventHandler<HTMLElement>) => void
+  onBlur?: (event: React.FocusEventHandler<HTMLElement>) => void
+  onMouseEnter?: (event: React.MouseEvent<HTMLElement>) => void
+  onMouseLeave?: (event: React.MouseEvent<HTMLElement>) => void
+  onKeyDown?: (event: React.KeyboardEvent<HTMLElement>) => void
+}
+
+const TOOLTIP_ARROW_EDGE_OFFSET = '16'
+
+const tooltipStyle = ({
+  direction,
+  noDelay,
+  align,
+  wrap,
+  open,
+}: Pick<Tooltip2Props, 'direction' | 'noDelay' | 'align' | 'wrap'> & {open: boolean}) => ({
+  position: 'relative',
+  display: 'inline-block',
+
+  '&::before': {
+    position: 'absolute',
+    zIndex: '1000001',
+    display: 'none',
+    content: '""',
+    width: 0,
+    height: 0,
+    color: 'neutral.onEmphasis',
+    pointerEvents: 'none',
+    borderStyle: 'solid',
+    borderWidth: '6px',
+    borderColor: 'transparent',
+    opacity: 0,
+    ...(open && {
```

Comment:
As a style refactor, what would you think about using data attributes for some of these states? This could avoid regenerating the objects for `styled-components` when props change as we could represent it like this statically:

```tsx
const StyledTooltip = styled.div`
  position: relative;
  display: inline-block;

  /* ... */

  [data-direction="n"], [data-direction="ne"], [data-direction="nw"] {
    /* ... */
  }
`
```

So that props will update `data-*` attributes but the styles won't have to be rebuilt each time.

Reply:
@joshblack I refactor the styles to use data attributes for `directions` prop and `open` state. I wasn't sure about `wrap` and `align` props - would that be too many data-attributes if I so the same thing for them? 🤔 cc @langermank would love your opinion on that too ☺️

Reply:
Hmm good question.. it might be nice to use data att for everything for the sake of consistency, and I wouldn't worry about having too many in there 😄

Reply:
Sounds good to me! 😄 I updated the code!